src/python/pants/pantsd/pailgun_server.py

```diff
@@ -87,8 +88,16 @@ def handle_error(self, exc=None):
     NailgunProtocol.send_exit_with_code(self.request, failure_code)
 
 
-class PailgunServer(TCPServer):
-  """A (forking) pants nailgun server."""
+class PailgunServer(ThreadingMixIn, TCPServer):
+  """A pants nailgun server.
+
+  This class spawns a thread per request via `ThreadingMixIn`: the thread body runs
+  `process_request_thread`, which we override.
+  """
+
+  # Override the ThreadingMixIn default, to minimize the chances of zombie pailgun processes.
+  daemon_threads = True
+  timeout = 0.05
```

Comment:
Perhaps order the timeout line above the comment to make it clear the timeout applies to just the daemon_threads line.